clouddq/classes/bigquery_client.py

```diff
@@ -202,3 +202,28 @@ def check_query_dry_run(self, query_string: str) -> None:
         except Forbidden as e:
             logger.error("User has insufficient permissions.")
             raise e
+
+    def is_table_exists(self, table: str) -> bool:
+        try:
+            client = self.get_connection()
+            client.get_table(table)
+            return True
+        except NotFound:
+            return False
+
+    def execute_query(self, query_string: str, job_config) -> bigquery.job.QueryJob:
```

Comment:
Please add type annotation for job_config, also could we make this an optional argument and provide a default job config if the caller doesn't specify a job_config?